### DISCUSSION TITLE
Enhancement Request: Allow Database Provider Configuration via Environment Variable

### DISCUSSION THREAD

**Discussion by Domains18** [prisma/prisma#24037] [Q&A]
## Problem
I frequently work with two database types, namely PostgreSQL and MySQL, in my development projects. While I can conveniently set the database URL via environment variables in my .env file, I encounter a challenge when it comes to configuring the database provider in the Prisma schema file (schema.prisma). Currently, I need to manually ensure that the database provider specified in the Prisma schema matches the one provided via the environment variable, which can be error-prone and inconvenient.

## Suggested solution
I propose enhancing Prisma to support configuring the database provider via an environment variable, similar to how the database URL is configured. This enhancement would streamline the development workflow, allowing users to seamlessly switch between different database types without needing to modify the Prisma schema file manually.
I honestly do not get why the `database  provider` cannot be allowed to use .env variables

**Comment by janpio** [comment]
We supported that in the past, but then removed it as it made it impossible to use our migration tooling. 

How would you expect the generated migration SQL to be handled in that case?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
